doc/userdoc/styleguide/styleguide.rst

```diff
@@ -0,0 +1,566 @@
+
+
+\``Double backticks\``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Use double backticks for
+
+
+  - inline code
+  - objects/functions
+  - model names
+  - NEST-specific vocabulary
+  - function call (e.g., ``nest.Create("iaf_psc_alpha")`` or  (``Create`` )
+  - Paths (e.g, You can find the models in ``nest-simulator/pynest/examples``)
+  - Key value pairs (``{key: value}``)
+  - Variables with assigned values ``x = 10``
+
+
+\`Single backticks\`
+~~~~~~~~~~~~~~~~~~~~
+
+Use single backticks for
+
+- Keys
+- Parameters
+- Variable names
+- Values
+
+but use double backticks when showing a complete example of variable with
+assigned value (e.g., \``volt = 37.0``)
+
+An example:
+
+::
+
+   Here we use the ``Create`` function to instantiate our model, in this case
+   ``iaf_psc_alpha``. We can modify the parameter `V_m` and set the value to
+   `50.0`.
+
+Rendered as
+
+
+   Here we use the ``Create`` function to instantiate our model, in this case
+   ``iaf_psc_alpha``. We can modify the parameters `V_m` and set the value to
+   `50.0`.
+
+\**Boldface\**
+~~~~~~~~~~~~~~
+
+If you want to emphasize a word or phrase in text, you can use **boldface**.
+
+NumPy style docstrings
+~~~~~~~~~~~~~~~~~~~~~~
+
+In the PyNEST code, we follow the rules for NumPy style docstrings as
+`explained here <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
+
+
+reStructuredText markup
+-----------------------
+
+Let us take a brief look into the markup of reStructuredText. You can find
+a full cheatSheet `here <https://thomas-cokelaer.info/tutorials/sphinx/
+rest_syntax.html>`_.
+
+.. note::
+
+   reStructuredText is sensitive to indentation. A single newline should always
+   be between any text and headings, tables, reference labels, or directives.
+   If the output format seems incorrect, double check the spaces and newlines
+   of the text.
+
+Code and code snippets
+~~~~~~~~~~~~~~~~~~~~~~
+
+For writing Python code, you can use the double colon
+
+
+ \::
+
+Rendered as
+
+   ::
+
+       python code here
+
+       x = 1
+
+       for i in line:
+           x = i
+           print(thing.x)
+
+For other code styles use the code-block directive (you can also use this one
+for python)
+
+
+
+\   .. code-block:: cpp
+
+Rendered as
+
+    .. code-block:: cpp
+
+       int main()
+       {
+         cout << "Hello, World!";
+         return 0;
+       }
+
+
+For input/output examples use the following syntax
+
+::
+
+   >>> input begins with 3, right-angled brackets
+   output is directly below input without any brackets
+
+
+Directives
+~~~~~~~~~~~
+
+This is not a complete guide to the directives of reStructuredText. For more
+options see `Sphinx directives <https://www.sphinx-doc.org/en/master/usage/
+restructuredtext/directives.html>`_ or `Docutils section on directives
+<http://docutils.sourceforge.net/docs/ref/rst/directives.html>`_.
+
+Equations
+^^^^^^^^^
+
+For equations, use LaTeX markup:
```

Comment:
Add information about inline math as well.

Reply:
This has been reworded